src/main/java/org/jitsi/srtp/crypto/MacAdapter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright @ 2020 - present 8x8, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.srtp.crypto;
+
+import org.bouncycastle.crypto.*;
+import org.bouncycastle.crypto.params.*;
+
+import javax.crypto.*;
+import javax.crypto.spec.*;
+import java.security.*;
+
+/**
+ * Adapts the <tt>javax.crypto.Mac</tt> class to the
+ * <tt>org.bouncycastle.crypto.Mac</tt> interface.
```

Comment:
Why do we need the adapter?

Reply:
Because we want to be able to use either JCE HMAC implementations or the BouncyCastle one, without exposing the difference to the calling code.  BouncyCastle's API is simpler, so I canonicalized to that.  (The alternative would be to use the BouncyCastle JCE provider, I guess?)

Reply:
>  (The alternative would be to use the BouncyCastle JCE provider, I guess?)

Yeah I think this is what I was wondering.  It doesn't give you the simpler API, I guess, but I'm wondering if we should go full on JCE and use providers or just use straight BC.

Reply:
BouncyCastle HMAC is slower than the JCE providers, which is why I made this change.  (Previously the code only used BC.)

Reply:
So is the issue that we're mixing (using BC in some places and JCE in others).  So we can't just install the BC provider and then use the generic functions to get the instances, as that would resolve everything to BC?

Reply:
I thought about this, but the thing is that our OpenSSLHmac implementation also implements the BouncyCastle Mac API, and it's sufficiently simpler than the JCE API (it's a simple light-weight interface rather than a complex set of things that expect to find a Provider underneath it) that this seemed simpler.

Reply:
Ok, you've got the best understanding and sounds like you've thought it over so I'm happy to defer to you here.